drivers/api/no_os_i2c.c

```diff
@@ -137,11 +139,16 @@ void no_os_i2cbus_remove(uint32_t bus_number)
 	struct no_os_i2cbus_desc *i2c_bus_desriptor = (struct no_os_i2cbus_desc *)
 			i2c_table[bus_number];
 
-	no_os_mutex_remove(i2c_bus_desriptor->mutex);
+	if (i2c_bus_desriptor->slave_number > 0)
```

Comment:
rename `desriptor` -> `descriptor` or delete it altogether - it makes for an unnecessarily long name, it could just as well be a variable named `bus`